### PR TITLE
feat: fail matchScreenshot when there is a visual diff to enable cypress retries config

### PR DESCRIPTION
**Pull request by aschwenn**
### :pencil: Description
the goal here is to enable consumers to use cypress's [test retries config](https://docs.cypress.io/guides/guides/test-retries)! currently, we only throw an error if cypress fails in some way to start up or take a screenshot. to be able to use retries, we need some nonzero exit code for tests where there is a visual diff, so that cypress understands that the test failed and retries it.

i'm modifying `match-screenshot` to now throw an error when there is a visual diff, but the action `run` script will count the number of "failed" commands and the number of `diff.png` images created. any successfully-run visual diff should create a `diff.png` file, so when the number of failed commands is greater than diffs created, we know that at least one visual test (non-diff) failure occurred!

testing locally showing that it retries successfully:
image

### :link: Related Issues
-

### CODE REVIEW

File: comparadise-utils/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
   },
   "dependencies": {
     "pixelmatch": "5.3.0",
-    "pngjs": "7.0.0"
+    "pngjs": "7.0.0",
+    "shared": "workspace:*"
```
Comment (danadajian) [inline comparadise-utils/package.json:22]: Will this work? 🤔

Comment (danadajian) [reply comparadise-utils/package.json:22]: Can you check if you build this package that the shared code gets bundled?

File: comparadise-utils/match-screenshot.ts [hunk 2]
```diff
@@ -115,13 +121,18 @@ export function matchScreenshot(
           'log',
           `❌ Actual image of ${name} differed by ${diffPixels} pixels.`
         );
+        visualDiffsExist = true;
```
Comment (danadajian) [inline comparadise-utils/match-screenshot.ts:124]: Why not exit right here?

Comment (aschwenn) [reply comparadise-utils/match-screenshot.ts:124]: i guess i was thinking it was iterating over something, but that's not the case

**Review by danadajian — APPROVED** [review]

**Pull request merged**